actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -18,16 +18,33 @@ def initialize(format)
     end
   end
 
+  class MissingSerializingFormat < KeyError
+    def initialize(format)
+      super "No serializing format defined for: '#{format}'."
+    end
+  end
+
   module Renderers
     extend ActiveSupport::Concern
 
     # A Set containing renderer names that correspond to available renderer procs.
     # Default values are <tt>:json</tt>, <tt>:js</tt>, <tt>:xml</tt>.
     RENDERERS = Set.new
+    FORMATTERS = Hash.new do |h, format|
```

Comment:
Is there a way to double check that we don't have duplicate format processors -- ex: should we prevent there from being 2 JSON formatters? I know we use a `Set` for `_renderers` 😬